data/datasource.go

```diff
@@ -447,6 +471,41 @@ func readBoltDB(source *Source, args ...string) ([]byte, error) {
 	return data, nil
 }
 
+func parsePluginArgs(pluginArgs []string) (map[string]plugin.Symbol, map[string]string, error) {
+	plugins := make(map[string]plugin.Symbol)
+	mediaTypes := make(map[string]string)
+	for _, v := range pluginArgs {
+		p, err := plugin.Open(v)
+		if err != nil {
+			return nil, nil, err
+		}
+		psm, err := p.Lookup(PluginSchemeFunction)
+		if err != nil {
+			return nil, nil, err
+		}
+		pluginSchemeName, err := psm.(func() (string, error))()
+		if err != nil {
+			return nil, nil, err
+		}
+		addSourceReader(pluginSchemeName, readPlugin)
+		pgm, err := p.Lookup(PluginGetFunction)
+		if err != nil {
+			return nil, nil, err
+		}
+		plugins[pluginSchemeName] = pgm
+		gmtsymbol, err := p.Lookup(PluginGetMediaTypeFunction)
+		if err != nil {
+			return plugins, nil, err
+		}
+		gmtfunc, ok := gmtsymbol.(func() string)
```

Comment:
This pattern of catching an ok on the function definition should be copied above.

Reply:
Also do we want this function to have the ability to error?